url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+	public static final int KW_USE=271;
+	public static final int KW_USER=272;
+	public static final int KW_USING=273;
+	public static final int KW_UTC=274;
+	public static final int KW_UTCTIMESTAMP=275;
+	public static final int KW_VALUE_TYPE=276;
+	public static final int KW_VARCHAR=277;
+	public static final int KW_VIEW=278;
+	public static final int KW_WHEN=279;
+	public static final int KW_WHERE=280;
+	public static final int KW_WHILE=281;
+	public static final int KW_WINDOW=282;
+	public static final int KW_WITH=283;
+	public static final int LCURLY=284;
+	public static final int LESSTHAN=285;
+	public static final int LESSTHANOREQUALTO=286;
+	public static final int LPAREN=287;
+	public static final int LSQUARE=288;
+	public static final int Letter=289;
+	public static final int MARCH=290;
+	public static final int MAY=291;
+	public static final int MINUS=292;
+	public static final int MINUTE=293;
+	public static final int MOD=294;
+	public static final int MONDAY=295;
+	public static final int MONTH=296;
+	public static final int NOTEQUAL=297;
+	public static final int NOVEMBER=298;
+	public static final int Number=299;
+	public static final int OCTOBER=300;
+	public static final int PLUS=301;
+	public static final int QUESTION=302;
+	public static final int RCURLY=303;
+	public static final int RPAREN=304;
+	public static final int RSQUARE=305;
+	public static final int RegexComponent=306;
+	public static final int SATURDAY=307;
+	public static final int SEMICOLON=308;
+	public static final int SEPTEMBER=309;
+	public static final int STAR=310;
+	public static final int SUNDAY=311;
+	public static final int SmallintLiteral=312;
+	public static final int StringLiteral=313;
+	public static final int THURSDAY=314;
+	public static final int TILDE=315;
+	public static final int TODAY=316;
+	public static final int TOMORROW=317;
+	public static final int TONIGHT=318;
+	public static final int TUESDAY=319;
+	public static final int TimeUnit=320;
+	public static final int TinyintLiteral=321;
+	public static final int WEDNESDAY=322;
+	public static final int WEEK=323;
+	public static final int WS=324;
+	public static final int YEAR=325;
+	public static final int YESTERDAY=326;
+
+	// delegates
+	// delegators
+	public Lexer[] getDelegates() {
+		return new Lexer[] {};
+	}
+
+	public HiveLexer() {} 
+	public HiveLexer(CharStream input) {
+		this(input, new RecognizerSharedState());
+	}
+	public HiveLexer(CharStream input, RecognizerSharedState state) {
+		super(input,state);
+	}
+	@Override public String getGrammarFileName() { return "HiveLexer.g"; }
+
+	// $ANTLR start "KW_TRUE"
+	public final void mKW_TRUE() throws RecognitionException {
+		try {
+			int _type = KW_TRUE;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:23:9: ( 'TRUE' )
+			// HiveLexer.g:23:11: 'TRUE'
+			{
+			match("TRUE"); 
+
+			}
+
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 369:384, 390:405, 411:426, 432:447, 453:468, 520:535, 541:556, 562:577, 583:597, 603:617, 623:637, 643:657, 663:678, 684:699, 705:720, 726:741, 747:762, 768:783, 789:804, 810:825, 831:846, 852:867, 873:888, 894:909, 915:930, 936:951, 957:972, 978:993, 999:1014, 1020:1035, 1041:1056, 1062:1077, 1083:1098, 1104:1119, 1125:1140, 1146:1161, 1167:1182, 1188:1203, 1209:1224, 1230:1245, 1251:1266, 1272:1287, 1293:1308, 1314:1329, 1335:1350, 1356:1371, 1377:1392, 1398:1413, 1419:1434, 1440:1455, 1461:1476, 1482:1497, 1503:1518, 1524:1539, 1545:1560, 1566:1581, 1587:1602, 1608:1623, 1629:1644, 1650:1665, 1671:1686, 1692:1707, 1713:1728, 1734:1749, 1755:1770, 1776:1791, 1797:1812, 1818:1833, 1839:1854, 1860:1875, 1881:1896, 1902:1917, 1923:1938, 1944:1959, 1965:1980, 1986:2001, 2007:2022, 2028:2043, 2049:2064, 2070:2085, 2091:2106, 2112:2127, 2133:2148, 2154:2169, 2175:2190, 2196:2211, 2217:2232, 2238:2253, 2259:2274, 2280:2295, 2301:2316, 2322:2337, 2343:2358, 2364:2379, 2385:2400, 2406:2421, 2427:2442, 2448:2463, 2469:2484, 2490:2505, 2511:2526, 2532:2547, 2553:2568, 2574:2589, 2595:2610, 2616:2631, 2637:2652, 2658:2673, 2679:2694, 2700:2715, 2721:2736, 2742:2757, 2763:2778, 2784:2799, 2805:2820, 2826:2841, 2847:2862, 2868:2883, 2889:2904, 2910:2925, 2931:2946, 2952:2967, 2973:2988, 2994:3009, 3015:3030, 3036:3051, 3057:3072, 3078:3093, 3099:3114, 3120:3135, 3141:3156, 3162:3177, 3183:3198, 3204:3219, 3225:3240, 3246:3261, 3267:3282, 3288:3303, 3309:3324, 3330:3345, 3351:3366, 3372:3387, 3393:3408, 3414:3429, 3435:3450, 3456:3471, 3477:3492, 3498:3513, 3519:3534, 3540:3555, 3561:3576, 3582:3597, 3603:3618, 3624:3639, 3645:3660, 3666:3681, 3687:3702, 3708:3723, 3729:3744, 3750:3765, 3771:3786, 3792:3807, 3813:3828, 3834:3849, 3855:3870, 3876:3891, 3897:3912, 3918:3933, 3939:3954, 3960:3975, 3981:3996, 4002:4017, 4023:4038, 4044:4059, 4065:4080, 4086:4101, 4107:4122, 4128:4143, 4149:4164, 4170:4185, 4191:4206, 4212:4227, 4233:4248, 4254:4269, 4275:4290, 4296:4311, 4317:4332, 4338:4353, 4359:4374, 4380:4395, 4401:4416, 4422:4437, 4443:4458, 4464:4479, 4485:4500, 4506:4521, 4527:4542, 4548:4563, 4569:4584, 4590:4605, 4611:4626, 4632:4647, 4653:4668, 4674:4689, 4695:4710, 4716:4731, 4737:4752, 4758:4773, 4779:4794, 4800:4815, 4821:4836, 4842:4857, 4863:4878, 4884:4899, 4905:4920, 4926:4941, 4947:4962, 4968:4983, 4989:5004, 5010:5025, 5031:5046, 5052:5067, 5073:5088, 5094:5109, 5115:5130, 5136:5151, 5157:5172, 5178:5193, 5199:5214, 5220:5235, 5241:5256, 5262:5277, 5283:5298, 5304:5319, 5325:5340, 5346:5361, 5367:5382, 5388:5403, 5409:5424, 5430:5445, 5451:5466, 5472:5487, 5493:5508, 5514:5529, 5535:5550, 5556:5571, 5577:5591, 5597:5611, 5617:5631, 5637:5651, 5657:5671, 5677:5691, 5697:5711, 5717:5731, 5737:5751, 5757:5771, 5828:5843, 5896:5911, 5917:5931, 5937:5952, 5958:5972, 5978:5992, 5998:6012, 6018:6032, 6038:6052, 6058:6072, 6078:6093, 6099:6113, 6119:6133, 6139:6153, 6159:6173, 6179:6193, 6199:6213, 9731:9746, 9961:9976, 9982:9997.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.